src/services/codefixes/convertToAsyncFunction.ts

```diff
@@ -0,0 +1,455 @@
+                }
+                else if (isArrowFunction(func)) {
+                    // if there is another outer dot then, don't actually return
+                    const innerRetStmts = getReturnStatementsWithPromiseCallbacks(createReturn(func.body as Expression));
+                    const innerCbBody = getInnerCallbackBody(checker, innerRetStmts, synthNamesMap, lastDotThenMap, context, prevArgName);
+                    if (innerCbBody.length > 0) {
+                        return createNodeArray(innerCbBody);
+                    }
+
+                    subtractReferences(func.body.pos, argName);
+                    if (hasPrevArgName && nextDotThen) {
+                        return createVariableDeclarationOrAssignment(prevArgName!, getSynthesizedDeepClone(func.body) as Expression);
+                    }
+                    else {
+                        return createNodeArray([createReturn(getSynthesizedDeepClone(func.body) as Expression)]);
+                    }
+                }
+                break;
+        }
+        return createNodeArray([]);
+    }
+
+    function removeReturns(stmts: NodeArray<Statement>, prevArgName: Identifier, varDeclFlags?: Map<NodeFlags | undefined>): NodeArray<Statement> {
+        const ret: Statement[] = [];
+        for (const stmt of stmts) {
+            if (isReturnStatement(stmt)) {
+                if (stmt.expression) {
+                    ret.push(createVariableStatement(/*modifiers*/ undefined,
+                        (createVariableDeclarationList([createVariableDeclaration(prevArgName, /*type*/ undefined, stmt.expression)], getNodeFlags(prevArgName, varDeclFlags)))));
+                }
+            }
+            else {
+                ret.push(getSynthesizedDeepClone(stmt));
+            }
+        }
+
+        return createNodeArray(ret);
+    }
+
+
+    function getInnerCallbackBody(checker: TypeChecker, innerRetStmts: Node[], synthNamesMap: Map<[Identifier, number]>, lastDotThenMap: Map<boolean>, context: CodeFixContextBase, prevArgName?: [Identifier, number]) {
+        let innerCbBody: Statement[] = [];
+        for (const stmt of innerRetStmts) {
+            forEachChild(stmt, function visit(node: Node) {
+                if (isCallExpression(node)) {
+                    const temp = parseCallback(node, checker, node, synthNamesMap, lastDotThenMap, context, prevArgName);
+                    innerCbBody = innerCbBody.concat(temp);
+                    if (innerCbBody.length > 0) {
+                        return;
+                    }
+                }
+                else if (!isFunctionLike(node)) {
+                    forEachChild(node, visit);
+                }
+            });
+        }
+        return innerCbBody;
+    }
+
+    function isCallback(node: CallExpression, funcName: string, checker: TypeChecker): boolean {

```

Comment:
This function seems to be doing two things, checking that the node is a PropertyAccessExpression with a given name, and 2. that the type of the whole expression is a Promise. 

First, it is not clear from the name that this is what it does, and second, do not see why the two should be lumped. 

in other places, you check independently for the type being a promise, or that it exists in the first place, so i would just remove that, and just call this function `isPropertyAccessExpressionWithName`